### PR TITLE
For lessons 04 write recipe and 05 write diagnostics

### PR DESCRIPTION
**Pull request by flicj191**
# Pull Request checklist

## Tasks

- [ ] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [ ] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

### CODE REVIEW

**Comment by flicj191** [comment]
@rbeucher  
i've pulled `gh-pages` into this to link to vscode setup page. Main changes from original were: 
- editing the example recipes in 'write your recipe' to use an ACCESS model
- loading xp65 module
- adding reference paths to cloned CMIP7-Hackathon folder - which has a branch called `exercise_example_files` where I've added some files
- added optional step in diagnostics lesson to copy the files to their folder for editing

File: _episodes/05-writingdiagnostics.md [hunk 1]
```diff
@@ -41,8 +40,8 @@ Let's get started!
 
 ## Understanding an existing Python diagnostic
 
-If you clone the ESMValTool repository, a folder called ``ESMValTool`` is
-created in your home/working directory.
+If you ran `check_hackathon`, a folder called ``ESMValTool`` is
+created in your working directory, this is a clone of the ESMValTool repository.
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
A clone of the ESMValTool repository should be available in your user folder in the `nf33`
`scratch` folder (`/scratch/nf33/$USER/ESMValTool`). If not, please make sure to run the
`check_hackathon` command after loading the `esmvaltool` module.
```

File: _episodes/05-writingdiagnostics.md [hunk 2]
```diff
@@ -51,7 +50,7 @@ directories:
 - *~/ESMValTool/esmvaltool/recipes/examples/recipe_python.yml*
 - *~/ESMValTool/esmvaltool/diag_scripts/examples/diagnostic.py*
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
```
ESMValTool/esmvaltool/recipes/examples/recipe_python.yml
ESMValTool/esmvaltool/diag_scripts/examples/diagnostic.py
```
```

File: _episodes/05-writingdiagnostics.md [hunk 3]
```diff
@@ -205,6 +204,32 @@ There are four main sections in the script:
...
+> Copy the files `diagnostic.py` and `recipe_python.yml` to your working folder
+> to keep the ones in the repo as templates unaltered while you can more easily 
+> find the files you are editing. Edit your recipe to point to your copy of 
+> `diagnostic.py`. Also, note the location for when you run your recipe.
+>
+>> ## Solution
+>>
+>> Example of your working folder: 
+>> - */scratch/.../Exercise3_files/recipe_python.yml*.
+>> - */scratch/.../Exercise3_files/diagnostic.py*
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
>```
>/scratch/nf33/$USER/Exercise3_files/recipe_python.yml
>/scratch/nf33/$USER/Exercise3_files/diagnostic.py
>```
```

File: _episodes/05-writingdiagnostics.md [hunk 4]
```diff
@@ -205,6 +204,32 @@ There are four main sections in the script:
...
+>>
+>> In your `recipe_python.yml`, edit the path to the diagnostic script.
+>> ```yaml
+>>     script1:
+>>       script: /scratch/.../Exercise3_files/diagnostic.py
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
>>       script: /scratch/nf33/$USER/Exercise3_files/diagnostic.py
```

File: _episodes/05-writingdiagnostics.md [hunk 5]
```diff
@@ -205,6 +204,32 @@ There are four main sections in the script:
...
+>>        quickplot:
+>> ```
+>> When running the recipe run to the full path of your recipe:
+>> ```bash
+>> esmvaltool run /scratch/.../Exercise3_files/recipe_python.yml
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
>> esmvaltool run /scratch/nf33/$USER/Exercise3_files/recipe_python.yml
```

File: _episodes/05-writingdiagnostics.md [hunk 6]
```diff
@@ -452,28 +481,28 @@ there:
...
->> ## Answer
+>> ## Solution
 >>
 >> In the recipe ``recipe_python.yml``, you could change ``plot_type`` and ``cmap``.
 >> As an example, we choose ``plot_type: pcolor`` and ``cmap: BuGn``:
 >>
 >> ```yaml
 >>     script1:
->>       script: examples/diagnostic.py
+>>       script: <path_to_script>/diagnostic.py
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
>>       script: <path_to_script diagnostic.py>
```

File: _episodes/05-writingdiagnostics.md [hunk 7]
```diff
@@ -452,28 +481,28 @@ there:
 
 ```yaml
      script1:
-       script: examples/diagnostic.py
+       script: <path_to_script>/diagnostic.py
```
Comment (rbeucher) [inline _episodes/05-writingdiagnostics.md]:
```suggestion
       script: <path_to_script diagnostic.py>
```

File: files/recipe_warming_stripes_additional_datasets.yml [hunk 8]
```diff
@@ -26,28 +29,28 @@ preprocessors:
...
-      temperature_anomalies_20th_century_london:
+      temperature_anomalies_20th_century_sydney:
         short_name: tas
-        preprocessor: anomalies_london
+        preprocessor: anomalies_sydney
         start_year: 1900
         end_year: 1999
         additional_datasets:
-          - {dataset: CanESM2, project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1}
+          - {dataset: ACCESS1-3, project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1}
```
Comment (rbeucher) [inline files/recipe_warming_stripes_additional_datasets.yml]: Can we do CMIP6 only?

**Review by rbeucher — CHANGES_REQUESTED** [review]
I have a few suggestions. See above. Good work!!

**Review by rbeucher — APPROVED** [review]

**Comment by rbeucher** [comment]
OK, I'm going to merge this for now. Might need some adjustments as we move forward.

**Pull request merged**